modules/core/src/main/java/org/apache/synapse/config/xml/endpoints/EndpointDefinitionSerializer.java

```diff
@@ -102,109 +102,140 @@ public void serializeEndpointDefinition(EndpointDefinition endpointDefinition,
             element.addChild(sec);
         }
 
-        if (endpointDefinition.getTimeoutAction() != SynapseConstants.NONE ||
-                endpointDefinition.getTimeoutDuration() > 0 || endpointDefinition.isDynamicTimeoutEndpoint()) {
+        if (!endpointDefinition.getTimeoutAction().equals("none") || endpointDefinition.isTimeoutActionDynamic() ||
+                isStringPositiveNumber(endpointDefinition.getTimeoutDuration())  || endpointDefinition.isDynamicTimeoutEndpoint()) {
 
             OMElement timeout = fac.createOMElement(
                     "timeout", SynapseConstants.SYNAPSE_OMNAMESPACE);
             element.addChild(timeout);
 
-            if (endpointDefinition.getTimeoutDuration() > 0 || endpointDefinition.isDynamicTimeoutEndpoint()) {
+            if (isStringPositiveNumber(endpointDefinition.getTimeoutDuration()) || endpointDefinition.isDynamicTimeoutEndpoint()) {
                 OMElement duration = fac.createOMElement(
                         "duration", SynapseConstants.SYNAPSE_OMNAMESPACE);
                 if (!endpointDefinition.isDynamicTimeoutEndpoint()) {
-                    duration.setText(Long.toString(endpointDefinition.getTimeoutDuration()));
+                    duration.setText(endpointDefinition.getTimeoutDuration());
                 } else {
-                    duration.setText('{' + endpointDefinition.getDynamicTimeoutExpression().getExpression() + '}');
+                    duration.setText('{' + endpointDefinition.getTimeoutDuration() + '}');
                 }
                 timeout.addChild(duration);
             }
 
-            if (endpointDefinition.getTimeoutAction() != SynapseConstants.NONE) {
+            if (!endpointDefinition.getTimeoutAction().equals("none") || endpointDefinition.isTimeoutActionDynamic()) {
                 OMElement action = fac.createOMElement("responseAction", SynapseConstants.SYNAPSE_OMNAMESPACE);
-                if (endpointDefinition.getTimeoutAction() == SynapseConstants.DISCARD) {
-                    action.setText("discard");
-                } else if (endpointDefinition.getTimeoutAction()
-                           == SynapseConstants.DISCARD_AND_FAULT) {
-                    action.setText("fault");
+                if (endpointDefinition.isTimeoutActionDynamic()) {
+                    action.setText('{' + endpointDefinition.getTimeoutAction() + '}');
+                } else {
+                    if (endpointDefinition.getTimeoutAction().equals("discard")) {
+                        action.setText("discard");
```

Comment:
Let's use the existing constants in all the places.

Reply:
updated